src/lib/signature_kind/compile_config/mina_signature_kind.ml

```diff
@@ -1,6 +1,6 @@
 include Mina_signature_kind_type
 
-let t =
+let t_DEPRECATED =
```

Comment:
You can also use
```
let t = [...] [@@deprecated]
```

Reply:
I'll add that.

My other motivation for the name change was to make it easier to tell during the refactoring how the value `Mina_signature_kind.t` was being used, as opposed to the type `Mina_signature_kind.t`. If that's too inelegant I can just have the `[@@deprecated]`.

Reply:
Is the `[@@deprecated]` supposed to do anything, like display an alert when compiling? I added it there and it did not seem to have an effect.

Reply:
I think it should display something. And I've run into these deprecation sometimes. I have no idea why it didn't work on your end though...

Reply:
The other `@@deprecated` notices in the code seem to be on the module declarations, like here

https://github.com/MinaProtocol/mina/blob/5f92160889019368ab38c667035cdb7a3b63d921/src/lib/logger/logger.mli#L172

Adding the deprecated tag to the `val t` in `mina_signature_kind.mli` makes using the value a fatal error, though.

Reply:
>Is the [@@deprecated] supposed to do anything, like display an alert when compiling? I added it there and it did not seem to have an effect.

Yes, it is supposed to make the compilation fails.
If you don'e have any error at compile time, it means we are ignoring the warning, and we must fix that.

Reply:
Does it make sense? Our goal is to rename `t` to `t_DEPRECATED` with a goal to fix it in a series of follow-up PRs that succeed this one shortly. Updating compilation flags to suppress the deprecated warning sounds overly complicated for something that we plan to be short-lived (1-2 weeks max) and removed completely soon.

Reply:
I'll at least remove the `[@@deprecated]` from the `let`, as that seems to be the wrong place for it.

Reply:
My apologies if my message has been considered as a request change. I mostly wanted to inform that the annotation deprecated did exist in OCaml if the solution was unknown to the author.

The current changes look good for the short term and I did not intend to request a change.